ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/AuthorizationMetaStoreFilterHook.java

```diff
@@ -43,7 +43,8 @@ public AuthorizationMetaStoreFilterHook(Configuration conf) {
   }
 
   @Override
-  public List<String> filterTableNames(String dbName, List<String> tableList) throws MetaException {
+  public List<String> filterTableNames(String catName, String dbName, List<String> tableList)
```

Comment:
filterTableMetas also needs to get implemented so that it goes through authorization filters (Ranger)

Reply:
The security model I'm proposing is that security is within a catalog, not across catalogs.  So Ranger etc would not have an opportunity to filter catalog names.  Everyone can always see all the catalogs that exist.  Whether they can see anything inside that catalog is up to the security configured for that catalog.  That's why I didn't implement a filterCatalogNames.